src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,10 @@
+package duke.command;
+import duke.task.TaskList;
+import duke.Ui;
+import duke.Storage;
+import duke.DukeException;
+
+public abstract class Command {
+    public abstract void execute(TaskList ls, Ui ui, Storage storage) throws DukeException;
+    public abstract boolean isExit();
```

Comment:
Since all but one of the commands have `isExit()` as `false`, this can be made into a non-abstract method.
```suggestion
    public boolean isExit() {
        return false;
    }
```